src/v/storage/tests/concat_segment_reader_test.cc

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright 2022 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+
+#include "model/tests/random_batch.h"
+#include "storage/directories.h"
+#include "storage/log_manager.h"
+#include "storage/tests/utils/disk_log_builder.h"
+#include "test_utils/tmp_dir.h"
+
+#include <seastar/testing/thread_test_case.hh>
+#include <seastar/util/defer.hh>
+
+constexpr size_t segment_size{1024};
```

Comment:
nit: use larger size to force `concat_segment_data_source_impl::get` method to read more than once from one segment (otherwise `input_stream::read` will read the entire segment in one go)